pkgs/servers/home-assistant/default.nix

```diff
@@ -383,6 +383,14 @@ in with py.pkgs; buildPythonApplication rec {
   preCheck = ''
     # the tests require the existance of a media dir
     mkdir /build/media
+
+    # error out when component test directory is missing, otherwise hidden by xdist execution :(
+    for component in ${lib.concatStringsSep " " (map lib.escapeShellArg componentTests)}; do
+      test -d "tests/components/$component" || {
+        >2& echo "ERROR: Tests for component '$component' were enabled, but they do not exist!"
+        exit 1
+      }
```

Comment:
Why not use ``if ; then ... fi`` here?

Reply:
What would be the benefit?

Reply:
if support running multiple commands without using {}.
```
      if [[ ! -d "tests/components/$component" ]]; then
        >2& echo "ERROR: Tests for component '$component' were enabled, but they do not exist!"
        exit 1
      fi
```